packages/react-dom/src/events/DOMEventProperties.js

```diff
@@ -52,104 +140,119 @@ const eventPriorities = new Map();
-  DOMTopLevelEventTypes.TOP_DRAG_END, 'dragEnd',
-  DOMTopLevelEventTypes.TOP_DRAG_START, 'dragStart',
-  DOMTopLevelEventTypes.TOP_DROP, 'drop',
-  DOMTopLevelEventTypes.TOP_FOCUS, 'focus',
-  DOMTopLevelEventTypes.TOP_INPUT, 'input',
-  DOMTopLevelEventTypes.TOP_INVALID, 'invalid',
-  DOMTopLevelEventTypes.TOP_KEY_DOWN, 'keyDown',
-  DOMTopLevelEventTypes.TOP_KEY_PRESS, 'keyPress',
-  DOMTopLevelEventTypes.TOP_KEY_UP, 'keyUp',
-  DOMTopLevelEventTypes.TOP_MOUSE_DOWN, 'mouseDown',
-  DOMTopLevelEventTypes.TOP_MOUSE_UP, 'mouseUp',
-  DOMTopLevelEventTypes.TOP_PASTE, 'paste',
-  DOMTopLevelEventTypes.TOP_PAUSE, 'pause',
-  DOMTopLevelEventTypes.TOP_PLAY, 'play',
-  DOMTopLevelEventTypes.TOP_POINTER_CANCEL, 'pointerCancel',
-  DOMTopLevelEventTypes.TOP_POINTER_DOWN, 'pointerDown',
-  DOMTopLevelEventTypes.TOP_POINTER_UP, 'pointerUp',
-  DOMTopLevelEventTypes.TOP_RATE_CHANGE, 'rateChange',
-  DOMTopLevelEventTypes.TOP_RESET, 'reset',
-  DOMTopLevelEventTypes.TOP_SEEKED, 'seeked',
-  DOMTopLevelEventTypes.TOP_SUBMIT, 'submit',
-  DOMTopLevelEventTypes.TOP_TOUCH_CANCEL, 'touchCancel',
-  DOMTopLevelEventTypes.TOP_TOUCH_END, 'touchEnd',
-  DOMTopLevelEventTypes.TOP_TOUCH_START, 'touchStart',
-  DOMTopLevelEventTypes.TOP_VOLUME_CHANGE, 'volumeChange',
+  TOP_CANCEL, 'cancel',
+  TOP_CLICK, 'click',
+  TOP_CLOSE, 'close',
+  TOP_CONTEXT_MENU, 'contextMenu',
+  TOP_COPY, 'copy',
+  TOP_CUT, 'cut',
+  TOP_AUX_CLICK, 'auxClick',
+  TOP_DOUBLE_CLICK, 'doubleClick',
+  TOP_DRAG_END, 'dragEnd',
+  TOP_DRAG_START, 'dragStart',
+  TOP_DROP, 'drop',
+  TOP_INPUT, 'input',
+  TOP_INVALID, 'invalid',
+  TOP_KEY_DOWN, 'keyDown',
+  TOP_KEY_PRESS, 'keyPress',
+  TOP_KEY_UP, 'keyUp',
+  TOP_MOUSE_DOWN, 'mouseDown',
+  TOP_MOUSE_UP, 'mouseUp',
+  TOP_PASTE, 'paste',
+  TOP_PAUSE, 'pause',
+  TOP_PLAY, 'play',
+  TOP_POINTER_CANCEL, 'pointerCancel',
+  TOP_POINTER_DOWN, 'pointerDown',
+  TOP_POINTER_UP, 'pointerUp',
+  TOP_RATE_CHANGE, 'rateChange',
+  TOP_RESET, 'reset',
+  TOP_SEEKED, 'seeked',
+  TOP_SUBMIT, 'submit',
+  TOP_TOUCH_CANCEL, 'touchCancel',
+  TOP_TOUCH_END, 'touchEnd',
+  TOP_TOUCH_START, 'touchStart',
+  TOP_VOLUME_CHANGE, 'volumeChange',
 ];
 
 const otherDiscreteEvents = [
-  DOMTopLevelEventTypes.TOP_CHANGE,
-  DOMTopLevelEventTypes.TOP_SELECTION_CHANGE,
-  DOMTopLevelEventTypes.TOP_TEXT_INPUT,
-  DOMTopLevelEventTypes.TOP_COMPOSITION_START,
-  DOMTopLevelEventTypes.TOP_COMPOSITION_END,
-  DOMTopLevelEventTypes.TOP_COMPOSITION_UPDATE,
+  TOP_CHANGE,
+  TOP_SELECTION_CHANGE,
+  TOP_TEXT_INPUT,
+  TOP_COMPOSITION_START,
+  TOP_COMPOSITION_END,
+  TOP_COMPOSITION_UPDATE,
 ];
 
-if (enableCreateEventHandleAPI) {
-  otherDiscreteEvents.push(
-    DOMTopLevelEventTypes.TOP_BEFORE_BLUR,
-    DOMTopLevelEventTypes.TOP_AFTER_BLUR,
+if (enableModernEventSystem) {
+  // Map "onFocus" to "onFocusIn" and "onBlur" to "onFocusOut"
+  discreteEventPairsForSimpleEventPlugin.push(
+    TOP_FOCUS_IN,
+    'focus',
+    TOP_FOCUS_OUT,
+    'blur',
+  );
```

Comment:
Should we also map `onFocusIn` -> "focusin" and `onFocusOut` -> "focusout" ?

Reply:
They are invalid events, you get a warning for using them.

Reply:
Right, but why? Seems a little odd that we *disallow* you to listen for "focusin" (via `onFocusIn`) when that's what we're really using anyway.

Reply:
It's the same for why we don't allow `onInput` but use it anyway behind the scenes right?

Reply:
I dunno. I mean, that also seems odd to me. 😄  I was just asking because it stood out. Your call on whether we should change it.